tests/test-update-refspec-endoflife.c

```diff
@@ -0,0 +1,243 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * Authors:
+ *  - Matthew Leeds <matthew.leeds@endlessm.com>
+ */
+
+#include <test-common/gpg.h>
+#include <test-common/misc-utils.h>
+#include <test-common/spawn-utils.h>
+#include <test-common/utils.h>
+
+#include <gio/gio.h>
+#include <locale.h>
+
+const gchar *next_ref = "REFv2";
+const OstreeCollectionRef _next_collection_ref = { (gchar *) "com.endlessm.CollectionId", (gchar *) "REFv2" };
+const OstreeCollectionRef *next_collection_ref = &_next_collection_ref;
+
+const OstreeCollectionRef _default_collection_ref_no_id = { NULL, (gchar *) "REF" };
+const OstreeCollectionRef *default_collection_ref_no_id = &_default_collection_ref_no_id;
+
+static GHashTable *
+create_eol_rebase_metadata (const gchar *ref_to_upgrade)
+{
+  GHashTable *ht = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
+  g_hash_table_insert (ht,
+                       g_strdup ("ostree.endoflife-rebase"),
```

Comment:
You can use `OSTREE_COMMIT_META_KEY_ENDOFLIFE_REBASE` for this.

Reply:
:+1: